CMakeLists.txt

```diff
@@ -417,37 +393,35 @@ if(${CMAKE_CXX_COMPILER_ID} MATCHES "GNU" OR
 
     # C++
     set(CMAKE_CXX_FLAGS_DEBUG ${BUILD_CXX_FLAGS_DEBUG}
-        CACHE STRING "Can't change here -- see BUILD_CXX..." FORCE)
+        CACHE INTERNAL "Control using BUILD_CXX")
     set(CMAKE_CXX_FLAGS_RELEASE ${BUILD_CXX_FLAGS_RELEASE}
-        CACHE STRING "Can't change here -- see BUILD_CXX..." FORCE)
+        CACHE INTERNAL "Control using BUILD_CXX")
     set(CMAKE_CXX_FLAGS_RELWITHDEBINFO ${BUILD_CXX_FLAGS_RELWITHDEBINFO}
-        CACHE STRING "Can't change here -- see BUILD_CXX..." FORCE)
+        CACHE INTERNAL "Control using BUILD_CXX")
     set(CMAKE_CXX_FLAGS_MINSIZEREL ${BUILD_CXX_FLAGS_MINSIZEREL}
-        CACHE STRING "Can't change here -- see BUILD_CXX..." FORCE)
+        CACHE INTERNAL "Control using BUILD_CXX")
 
     # C
     set(CMAKE_C_FLAGS_DEBUG ${BUILD_C_FLAGS_DEBUG}
-        CACHE STRING "Can't change here -- see BUILD_C..." FORCE)
+        CACHE INTERNAL "Control using BUILD_CXX")
     set(CMAKE_C_FLAGS_RELEASE ${BUILD_C_FLAGS_RELEASE}
-        CACHE STRING "Can't change here -- see BUILD_C..." FORCE)
+        CACHE INTERNAL "Control using BUILD_CXX")
     set(CMAKE_C_FLAGS_RELWITHDEBINFO ${BUILD_C_FLAGS_RELWITHDEBINFO}
-        CACHE STRING "Can't change here -- see BUILD_C..." FORCE)
+        CACHE INTERNAL "Control using BUILD_CXX")
     set(CMAKE_C_FLAGS_MINSIZEREL ${BUILD_C_FLAGS_MINSIZEREL}
-        CACHE STRING "Can't change here -- see BUILD_C..." FORCE)
+        CACHE INTERNAL "Control using BUILD_CXX")
 
     set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake" ${CMAKE_MODULE_PATH})
     if(SIMBODY_COVERAGE)
         find_package(codecov)
     endif()
-
-endif()
-
-## When building in any of the Release modes, tell VC++ cl compiler to use
-## intrinsics (i.e. sqrt instruction rather than sqrt subroutine) with
-## flag /Oi.
+elseif(MSVC)
 ## Caution: can't use CMAKE_CXX_COMPILER_ID MATCHES MSVC here because
 ## "MSVC" is a predefined CMAKE variable and will get expanded to 1 or 0
-if(MSVC)
```

Comment:
I'm not sure if its possible to use GCC/Clang within MSVC (i.e. definitely mutually exclusive, therefore combining the if/elseif is correct)? In any case, the flag setting contents assumes the use of VC++ cl compiler.